homeassistant/components/sonos/media_player.py

```diff
@@ -619,6 +624,11 @@ def update(self, now=None):
         except SoCoException:
             pass
 
+    @callback
+    def async_update_media(self, event=None):
+        """Update information about currently playing media."""
+        self.hass.async_add_job(self.update_media, event)
```

Comment:
We should use `self.hass.async_add_executor_job` to schedule on the executor.

Reply:
This is being called back from `pysonos` which can only take a callback so we can't `await` here.

Reply:
But it doesn't need to be awaited because that's all add job does under the hood anyways.  Will adjust.

Surprisingly makes more sense when not looking at it after midnight and getting sleep 🤦‍♂️

Reply:
Resolved in https://github.com/home-assistant/core/pull/48684 (but docker hub is broken at the moment so the CI is likely to fail)